File: Frontend/apps/astrogame_app/lib/models/players/player.dart
import 'package:astrogame_app/communications/converters/guid_converter.dart';
import 'package:astrogame_app/models/common/guid.dart';

import 'package:json_annotation/json_annotation.dart';

import 'colonized_stellar_object.dart';
import 'player_species.dart';

part 'player.g.dart';

@GuidConverter()
@JsonSerializable()
class Player {
  late Guid id;
  late String username;
  late Guid playerSpeciesId;
  late List<ColonizedStellarObject> colonizedObjects;
  late PlayerSpecies? playerSpecies;

  Player();

  factory Player.fromJson(Map<String, dynamic> json) => _$PlayerFromJson(json);
  Map<String, dynamic> toJson() => _$PlayerToJson(this);
}

File: Backend/AstroGame.Shared/Models/Players/Player.cs
using System;
using System.Collections.Generic;

namespace AstroGame.Shared.Models.Players
{
    public class Player
    {
        public Guid Id { get; set; }
        public string Username { get; set; }
        public string ConfirmationToken { get; set; }

        public Guid? PlayerSpeciesId { get; set; }
        
        //TODO: public Guid AllianceId { get; set; }

        public Credentials Credentials { get; set; }
        public List<ColonizedStellarObject> ColonizedObjects { get; set; }
        public PlayerSpecies PlayerSpecies { get; set; }
    }
}